### ISSUE TITLE
Generelized Omega - wrong SS?

### ISSUE THREAD

**Issue by mattansb** [doomlab/MOTE#8]
In you [website](https://www.aggieerin.com/shiny-server/tests/gosrmss.html) you specify Omega<sup>2</sup><sub>G</sub> as:

![image](https://user-images.githubusercontent.com/35330040/96366398-1dcd9400-1150-11eb-9fe0-5711fe3d726a.png)

However, in the orignal paper by Olejnik and Algina, 
it seems like the denominator should be [SS<sub>A</sub> + SS<sub>Other</sub>] 
where you have [SS<sub>***Total***</sub> + SS<sub>Other</sub>]:

![image](https://user-images.githubusercontent.com/35330040/96366439-4a81ab80-1150-11eb-837b-19e696a616af.png)

If this isn't a mistake, can you please help me understand what I'm missing here?
Thanks!

**Comment by doomlab** [comment]
@mattansb - just finally dug this out of my email. We are using the formula on that same page (switching A and B on our explanation). 

![image](https://user-images.githubusercontent.com/27783894/99453140-81371700-28f2-11eb-9a31-9182727b4668.png) 

The table you've included is for three way designs with various mixed factors - we note in our guide that we only are doing two-way designs (maybe this should be clearer, but you only enter information for two of them).

**Comment by mattansb** [comment]
Okay, I see now!

However this raises other concerns - looking at the paper, from page 441-442, it seem that this formula is only applicable for:
- A **fixed effects** model 
  - With only 2 IVs
  - With one IV manipulated and the other measured
- The generalized Omega in the function is only applicable for the main effect of **measured** IV.

Whereas the docs of the function seem to claim it is applicable for both fixed and mixed models (and it is suggested that there is something called a partial generalized omega squared? *"Remember if you have two or more IVs, these values are partial omega squared."*)...

Unless again I am missing something? 😅

**Comment by mattansb** [comment]
Likewise, the `ges.partial.SS.mix()` seems to be for Partial Generalized Eta-Squared, but I could not find a reference to such an effect size outside your package? And the formula seems also to only be applicable for **manipulated** IVs?

And I'm not sure where `ges.partial.SS.rm()` is taken from?

Sorry for the deep dive here...

**Comment by doomlab** [comment]
@mattansb you are raising an excellent point that I have been thinking about. I know that the generalized eta/omega paper talks a lot about manipulated versus measured ... but I'm not 100% convinced this distinction at all matters for the estimation of the effect. Obviously, the data collection procedure does (i.e., between versus repeated). I just haven't connected all the dots on how to show this point exactly. Either way some clarifying language is in order. 

Everything generalized (eta/omega) is Olejnik and Algina - I'm going to mark these notes to make sure we update the citation information online before I start working on the paper for this thing.

**Comment by mattansb** [comment]
The distinction between manipulated and observed is, to my understanding, the whole point of the generalized measures; The generalized effect size tries to determine the effect size we would see for this "term" if it was to be set outside the lab, in the wild, so it matters if a IV varies in the wild (e.g., birth order) or if it doesn't (e.g., treatment group).

Here is an example from the `afex` package showing how the generalized effect size is affected by not only if the term of interest is labelled "observed" or not, but also by how *other* terms are labelled.

``` r
library(afex)

data(obk.long, package = "afex")

a <- aov_4(value ~ treatment * gender * phase * hour +
             (phase * hour | id),
           data = obk.long)
```

The `ges` column is the generalized Eta squared.
Focusing on the `gender` effect.

When non observed, we get `ges = 0.15`.
``` r
anova(a)[2, ]
#> Anova Table (Type 3 tests)
#> 
#> Response: value
#>        num Df den Df    MSE      F    ges Pr(>F)  
#> gender      1     10 22.806 3.6591 0.1516 0.0848 .
#> ---
#> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
```

When only `gender` is observed, we get `ges = 0.11`.
```R
anova(a, observed = "gender")[2, ] # gender is observed
#> Anova Table (Type 3 tests)
#> 
#> Response: value
#>        num Df den Df    MSE      F     ges Pr(>F)  
#> gender      1     10 22.806 3.6591 0.11481 0.0848 .
#> ---
#> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
```
When `gender` is not observed, but `treatment` is, we get `ges = 0.08`.
```R
anova(a, observed = "treatment")[2, ] # something else (not gender) is observed
#> Anova Table (Type 3 tests)
#> 
#> Response: value
#>        num Df den Df    MSE      F      ges Pr(>F)  
#> gender      1     10 22.806 3.6591 0.085304 0.0848 .
#> ---
#> Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
```

This is just a quick example. A situation can also be manufactured where the differences are much larger.

**Comment by doomlab** [comment]
@mattansb I get that we can calculate them differently, I'm just not convinced that this distinction is necessary (or educated enough on why we think this difference is necessary). It seems to me that effect sizes (and their distributions) should be determined by the underlying effect ... why should the effect care if they are measured or manipulated? 

I obviously can be uninformed here, if you have a reference (or I just have forgotten everything but the math from the main paper we are talking about).

**Comment by mattansb** [comment]
> why should the effect care if they are measured or manipulated?

This is *exactly* the point of the generalized eta/omega. From the paper's abstract:

![image](https://user-images.githubusercontent.com/35330040/104572895-edb56480-565c-11eb-99ff-ca40030a0774.png)

So what the GES or GOS do is "account" for the design, and allow to compare the effect sizes of a term across different designs.
Else you can just report the partial / non-partial eta/omega.

Here is a short comp between the different eta/omegas (from [Lakens, 2013](https://doi.org/10.3389/fpsyg.2013.00863))

![image](https://user-images.githubusercontent.com/35330040/104573185-4a188400-565d-11eb-9e82-9834881355e5.png)

**Comment by mattansb** [comment]
This is also why it doesn't make sense to have a partial-generalized eta/omega, as you need to account for *the whole design* (not just *part* of it).

Hope this is helping / making sense?

**Comment by doomlab** [comment]
@mattansb alright, I took some time this morning to reread the original paper. I don't know if I misread the original, didn't comprehend it correctly, or what but I get what is going on now. We were sort of talking about two different issues, but both things that need to be addressed. Unfortunately, that's going to take some rewriting of the ges related functions and some thinking on how best to explain this because I don't think this measured/manipulated design distinction is necessarily clear.

**Comment by mattansb** [comment]
> because I don't think this measured/manipulated design distinction is necessarily clear.

Yes, I agree - I think it is better (probably) to frame this as "addressing bias in effect size due to study design".

Glad I brought this up (:

(On a side note, we've been trying to implement gen-Omega in [`effectsize`](https://easystats.github.io/effectsize/) for a while, but have found it *very* hard to do! The equations are hard to generalize (pun intended!))
